lib/graphql/execution/interpreter/runtime.rb

```diff
@@ -473,14 +468,29 @@ def set_result(selection_result, result_name, value)
                 set_result(parent, name_in_parent, nil)
                 # This is odd, but it's how it used to work. Even if `parent` _would_ accept
                 # a `nil`, it's marked dead. TODO: check the spec, is there a reason for this?
```

Comment:
This one just seems wrong to me. If the parent accepts a `nil` it shouldn't be marked dead, and if it doesn't then we should rely on the error bubbling up and marking it dead at that point.

Reply:
(Addressed by #3537)